src/main/java/org/broadinstitute/hellbender/utils/runtime/ScriptExecutor.java

```diff
@@ -0,0 +1,142 @@
+package org.broadinstitute.hellbender.utils.runtime;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Base class for executors that find and run scripts in an external script engine process (R, Python, etc).
+ *
+ * Subclasses must implement:
+ *
+ *   {@link #getApproximateCommandLine}
+ *   {@link #getScriptException}
+ */
+public abstract class ScriptExecutor<T extends ScriptExecutorException> {
```

Comment:
As far as I can tell this type variable is unnecessary.  Subclasses can already override with a more specific return type without making it a parameterized type.  Maybe I'm missing something more subtle though.